src/matrix.rs

```diff
@@ -106,10 +127,80 @@ impl<T:Add<T, Output=T> + Copy + Clone + ApproxEq<T> + Mul<T, Output=T> + One +
 
         return self.mul(&matrix);
     }
+
+    pub fn create_translation(x: T, y: T, z: T) -> Matrix4<T> {
+        let (_0, _1): (T, T) = (Zero::zero(), One::one());
+        Matrix4(_1.clone(), _0.clone(), _0.clone(), _0.clone(),
+                _0.clone(), _1.clone(), _0.clone(), _0.clone(),
+                _0.clone(), _0.clone(), _1.clone(), _0.clone(),
+                         x,          y,          z, _1.clone())
+    }
+
+    pub fn create_scale(x: T, y: T, z: T) -> Matrix4<T> {
+        let (_0, _1): (T, T) = (Zero::zero(), One::one());
+        Matrix4(         x, _0.clone(), _0.clone(), _0.clone(),
+                _0.clone(),          y, _0.clone(), _0.clone(),
+                _0.clone(), _0.clone(),          z, _0.clone(),
+                _0.clone(), _0.clone(), _0.clone(), _1.clone())
+    }
+
+    pub fn create_rotation(x: T, y: T, z: T, theta: T) -> Matrix4<T> {
+        let _0: T = Zero::zero();
+        let _1: T = One::one();
+        let _2: T = NumCast::from(2).unwrap();
+        let _half: T = NumCast::from(0.5).unwrap();
+
+        let xx = x * x;
+        let yy = y * y;
+        let zz = z * z;
+
+        let half_theta = theta * _half;
+        let sc = half_theta.sin() * half_theta.cos();
+        let sq = half_theta.sin() * half_theta.sin();
+
+        Matrix4(
+            _1.clone() - _2.clone() * (yy + zz) * sq,
```

Comment:
I'm going to assume that 1 - 2 (y² + z²) sin θ = x² (1 - cos θ) + cos θ (comparing against [glRotate]), and likewise for the others on the diagonal.

[glRotate]: https://www.opengl.org/sdk/docs/man2/xhtml/glRotate.xml